### ISSUE TITLE
JsDoc import - Doesnt work

### ISSUE THREAD

**Issue by gcaseres** [vuejs/vetur#999]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: Linux
- Vetur version: 0.14.2
- VS Code version: 1.29.1

## Problem

In a Javascript project, I want to import JsDoc types defined in another file.
I'm using the following syntax:

```@typedef {import('route_to_imported_file').TypeDefName} TypeDefName```

It looks like Vetur isnt able to resolve the imported module. Intellisense doesnt work.

But it works if I do it from a common .js file.

## Reproducible Case
[https://github.com/gcaseres/veturpack-json_import_issue](https://github.com/gcaseres/veturpack-json_import_issue)

Intellisense works well in ```app.js``` (common js file)
But the same jsdoc import doesn't work in ```components/Counter.vue``` (vue file)

**Comment by mqdev** [comment]
Same here, just after the `script`:

```
/**
 * @typedef {import('../../types.js').Livro} Livro
 */
```

Vetur server output:
```Vetur initialized
c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:4238
            throw e;
            ^

Error: Debug Failure. Found typedef tag attached to node which it should not be!
    at nameForNamelessJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12345:26)
    at getNameOfJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12353:36)
    at Object.getNameOfDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12389:24)
    at Object.hasDynamicName (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:10388:23)
    at declareSymbol (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21097:33)
    at declareModuleMember (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21231:28)
    at bindBlockScopedDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22381:25)
    at bindWorker (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22807:32)
    at bind (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22558:13)
    at bindJSDocTypedefTagIfAny (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22595:25)
[Info  - 17:50:36] Connection to server got closed. Server will restart.
Vetur initialized
c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:4238
            throw e;
            ^

Error: Debug Failure. Found typedef tag attached to node which it should not be!
    at nameForNamelessJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12345:26)
    at getNameOfJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12353:36)
    at Object.getNameOfDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12389:24)
    at Object.hasDynamicName (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:10388:23)
    at declareSymbol (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21097:33)
    at declareModuleMember (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21231:28)
    at bindBlockScopedDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22381:25)
    at bindWorker (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22807:32)
    at bind (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22558:13)
    at bindJSDocTypedefTagIfAny (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22595:25)
[Info  - 17:50:39] Connection to server got closed. Server will restart.
Vetur initialized
c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:4238
            throw e;
            ^

Error: Debug Failure. Found typedef tag attached to node which it should not be!
    at nameForNamelessJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12345:26)
    at getNameOfJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12353:36)
    at Object.getNameOfDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12389:24)
    at Object.hasDynamicName (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:10388:23)
    at declareSymbol (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21097:33)
    at declareModuleMember (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21231:28)
    at bindBlockScopedDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22381:25)
    at bindWorker (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22807:32)
    at bind (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22558:13)
    at bindJSDocTypedefTagIfAny (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22595:25)
[Info  - 17:50:42] Connection to server got closed. Server will restart.
Vetur initialized
c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:4238
            throw e;
            ^

Error: Debug Failure. Found typedef tag attached to node which it should not be!
    at nameForNamelessJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12345:26)
    at getNameOfJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12353:36)
    at Object.getNameOfDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12389:24)
    at Object.hasDynamicName (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:10388:23)
    at declareSymbol (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21097:33)
    at declareModuleMember (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21231:28)
    at bindBlockScopedDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22381:25)
    at bindWorker (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22807:32)
    at bind (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22558:13)
    at bindJSDocTypedefTagIfAny (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22595:25)
[Info  - 17:50:44] Connection to server got closed. Server will restart.
[Error - 17:50:44] Request textDocument/hover failed.
Error: Connection got disposed.
    at Object.dispose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\main.js:876:25)
    at Object.dispose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:71:35)
    at LanguageClient.handleConnectionClosed (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:2150:42)
    at LanguageClient.handleConnectionClosed (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\main.js:150:15)
    at closeHandler (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:2137:18)
    at CallbackList.invoke (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:62:39)
    at Emitter.fire (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:120:36)
    at closeHandler (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\main.js:226:26)
    at CallbackList.invoke (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:62:39)
    at Emitter.fire (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:120:36)
    at IPCMessageReader.fireClose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
    at ChildProcess.constructor.eventEmitter.on (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\messageReader.js:213:45)
    at ChildProcess.emit (events.js:187:15)
    at maybeClose (internal/child_process.js:961:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
Vetur initialized
c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:4238
            throw e;
            ^

Error: Debug Failure. Found typedef tag attached to node which it should not be!
    at nameForNamelessJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12345:26)
    at getNameOfJSDocTypedef (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12353:36)
    at Object.getNameOfDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:12389:24)
    at Object.hasDynamicName (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:10388:23)
    at declareSymbol (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21097:33)
    at declareModuleMember (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:21231:28)
    at bindBlockScopedDeclaration (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22381:25)
    at bindWorker (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22807:32)
    at bind (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22558:13)
    at bindJSDocTypedefTagIfAny (c:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\server\node_modules\typescript\lib\typescript.js:22595:25)
[Error - 17:50:47] Connection to server got closed. Server will not be restarted.
[Error - 17:50:47] Request textDocument/hover failed.
Error: Connection got disposed.
    at Object.dispose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\main.js:876:25)
    at Object.dispose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:71:35)
    at LanguageClient.handleConnectionClosed (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:2150:42)
    at LanguageClient.handleConnectionClosed (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\main.js:150:15)
    at closeHandler (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-languageclient\lib\client.js:2137:18)
    at CallbackList.invoke (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:62:39)
    at Emitter.fire (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:120:36)
    at closeHandler (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\main.js:226:26)
    at CallbackList.invoke (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:62:39)
    at Emitter.fire (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\events.js:120:36)
    at IPCMessageReader.fireClose (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
    at ChildProcess.constructor.eventEmitter.on (C:\Users\2omma\.vscode\extensions\octref.vetur-0.16.2\node_modules\vscode-jsonrpc\lib\messageReader.js:213:45)
    at ChildProcess.emit (events.js:187:15)
    at maybeClose (internal/child_process.js:961:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)```

**Comment by octref** [comment]
This seems to be solved in TS 3.3, which you can get by:

- `yarn add -D typescript@3.3` in workspace
-  `"vetur.useWorkspaceDependencies": true`

See #682 for details.

![image](https://user-images.githubusercontent.com/4033249/54790548-b954f680-4bf3-11e9-9601-b873d5f46c35.png)

This issue will be completely gone after #1163.

**Comment by mqdev** [comment]
Confirming, after update it's working!
